tests/safeds_runner/server/test_websocket_mock.py

```diff
@@ -15,6 +17,12 @@
     message_type_runtime_error,
     message_type_runtime_progress,
 )
+from safeds_runner.server.pipeline_manager import PipelineManager
+
+
+@pytest.fixture(scope="session", autouse=True)
+def _prepare_tests() -> None:
+    safeds_runner.server.main.app_pipeline_manager = PipelineManager()
```

Comment:
This is quite ugly. It's also problematic that `PipelineManager` must never be initialized on the global level of a module - a restriction that must be documented carefully.

I'd instead suggest removing the problematic code from the `__init__` method of the `PipelineManager`. Instead, lazily initialize the fields such as `_multiprocessing_manager` when they are needed, like when `execute_pipeline` is called. You can turn the attributes into [`cached_properties`](https://docs.python.org/3/library/functools.html#functools.cached_property) for this purpose.

Reply:
The message queue thread still needs to be started at some point.
Should this also be delayed until the first call to `execute_pipeline` was made?

Reply:
Is there some downside to delaying it?

Reply:
There should be none.